tests/test_core.py

```diff
@@ -2383,6 +2383,18 @@ def test_switch_issue_913_using_integers():
     common(d, b"\xab", 171, 1, x=1)
     common(d, b"\x09\x00", 9, 2, x=2)
 
+
+def test_lazy_rebuild():
+    d = Struct(
+        "foo" / Int32ul,
+        "bar" / Rebuild(Int32ul, lambda ctx: ctx.baz),
+        "baz" / Rebuild(Int32ul, lambda ctx: ctx.foo),
+    )
+    obj = {"foo": 4, "bar": lambda ctx: ctx.baz, "baz": lambda ctx: ctx.foo}
```

Comment:
Why isn't this just `{"foo": 4}`? Asked differently: Why are "bar" and "baz" functions?

I think it's a bit hacky to allow the *data* to decide how it should be processed (~ functions). I'm not sure if there's a precedent for this in construct.

Instead, the `lambda ctx: ctx.baz` in `Rebuild` should be deferred until `ctx.baz` is available.

Is there an actual benefit doing it the proposed way (allowing functions in input to `build`) over deferring `Rebuild`?